packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -77,6 +77,17 @@ const bAddLightContent = esTemplate`
     });
 `<EsCallExpression>;
 
+function getShadowSlottedContent(slottableChildren: IrChildNode[], cxt: TransformerContext) {
+    return optimizeAdjacentYieldStmts(
+        slottableChildren.flatMap((child) => {
```

Comment:
This could lead to subtle bugs with `nextSibling`/`prevSibling`, which are set in `irChildrenToEs`. (We fixed a bug a week or so ago that was because we had `flatMap` + `irToEs` rather than `irChildrenToEs`, but I don't remember which PR it was.)

https://github.com/salesforce/lwc/blob/ee2edae09f6e080ccc92703864a463e31b4bb3b3/packages/%40lwc/ssr-compiler/src/compile-template/ir-to-es.ts#L69-L78

Reply:
`nextSibling` will probably be there for the foreseeable future, because we need it for adjacent text node concatenation.

I think Will is right, and we may need to create a helper method for cases where we need to `flatMap`. For example if you have adjacent text nodes in your tests (`{foo}{bar}`) which resolve to the empty string / undefined, then you should be able to repro problems with `nextSibling`.

Reply:
Thanks @nolanlawson . The obvious way for a helper like that would be to add an optional callback to `irChildrenToEs` which hooks into its own `flatMap` callback. I'll try that once #5036 is merged.